src/main/java/io/fabric8/elasticsearch/plugin/discovery/kubernetes/KubernetesDiscoveryPlugin.java

```diff
@@ -53,12 +53,13 @@ public static boolean isDiscoveryAlive(Settings settings, ESLogger logger) {
       return false;
     }
 
-    if (!checkProperty(KubernetesAPIService.Fields.NAMESPACE, settings.get(KubernetesAPIService.Fields.NAMESPACE), logger) ||
-      !checkProperty(KubernetesAPIService.Fields.SERVICE_NAME, settings.get(KubernetesAPIService.Fields.SERVICE_NAME), logger)) {
+    if ( !hasNamespace(settings, logger) || !(hasServiceName(settings, logger) ^ hasPodLabel(settings, logger)) ) {
```

Comment:
The '^' operater seems like a strange syntax here and is equivalent to single '|'  (I think)